tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -337,54 +298,102 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 				samples: [],
 				variance: Number.NaN,
 			};
+
 			do {
 				await testObject.beforeIteration?.();
-
 				global.gc();
-				benchmarkStats.samples.before.memoryUsage.push(process.memoryUsage());
-				benchmarkStats.samples.before.heap.push(v8.getHeapStatistics());
-				benchmarkStats.samples.before.heapSpace.push(v8.getHeapSpaceStatistics());
+				(
+					benchmarkStats.customData["Samples used"] as MemorySampleData
+				).before.memoryUsage.push(process.memoryUsage());
+				(benchmarkStats.customData["Samples used"] as MemorySampleData).before.heap.push(
+					v8.getHeapStatistics(),
+				);
+				(
+					benchmarkStats.customData["Samples used"] as MemorySampleData
+				).before.heapSpace.push(v8.getHeapSpaceStatistics());
 
 				global.gc();
 				await testObject.run();
 
 				await testObject.afterIteration?.();
 
 				global.gc();
-				benchmarkStats.samples.after.memoryUsage.push(process.memoryUsage());
-				benchmarkStats.samples.after.heap.push(v8.getHeapStatistics());
-				benchmarkStats.samples.after.heapSpace.push(v8.getHeapSpaceStatistics());
-
-				benchmarkStats.runs++;
+				(
+					benchmarkStats.customData["Samples used"] as MemorySampleData
+				).after.memoryUsage.push(process.memoryUsage());
+				(benchmarkStats.customData["Samples used"] as MemorySampleData).after.heap.push(
+					v8.getHeapStatistics(),
+				);
+				(
+					benchmarkStats.customData["Samples used"] as MemorySampleData
+				).after.heapSpace.push(v8.getHeapSpaceStatistics());
+
+				benchmarkStats.customData.Iterations =
+					(benchmarkStats.customData.Iterations as number) + 1;
 
 				const heapUsedArray: number[] = [];
-				for (let i = 0; i < benchmarkStats.samples.before.memoryUsage.length; i++) {
+				for (
+					let i = 0;
+					i <
+					(benchmarkStats.customData["Samples used"] as MemorySampleData).before
+						.memoryUsage.length;
+					i++
+				) {
 					heapUsedArray.push(
-						benchmarkStats.samples.after.memoryUsage[i].heapUsed -
-							benchmarkStats.samples.before.memoryUsage[i].heapUsed,
+						(benchmarkStats.customData["Samples used"] as MemorySampleData).after
+							.memoryUsage[i].heapUsed -
+							(benchmarkStats.customData["Samples used"] as MemorySampleData).before
+								.memoryUsage[i].heapUsed,
 					);
 				}
 				heapUsedStats = getArrayStatistics(heapUsedArray, options.samplePercentageToUse);
 
 				// Break if max elapsed time passed, only if we've reached the min sample count
 				if (
-					benchmarkStats.runs >= options.minSampleCount &&
+					(benchmarkStats.customData.Iterations as number) >= options.minSampleCount &&
```

Comment:
Similar to above, let's use local variables of known types to do all the work inside the runner, and only at the end assign it to `benchmarkStats.customData["<something>"]` to avoid all this casting in the middle of things. It makes the code noticeably harder to read.

Reply:
Synced offline. Should covered this one.